### DISCUSSION TITLE
Is there a way to get a list of active listings with images in a single API call without OAuth?

### DISCUSSION THREAD

**Discussion by lubna** [etsy/open-api#1014] [Q&A]
Hi,

I'm trying to build a landing page for a friend that shows a list of their Etsy listings on their website and links back to the listing on Etsy so people can buy it from there. It's just another way for them to get people to their Etsy store and I don't want to integrate any of the purchase functionality into the site.

The `findAllActiveListingsByShop` endpoint gets me a list of all the active items in the shop and only requires an API token, however the results doesn't return any images (I only need the main one since the products are very visual). It looks like there is a way to request images to be included on the `getListingsByShop` endpoint but that requires OAuth and I don't want people to have to sign in just to see the listings. I tried using the `findAllActiveListingsByShop` and then calling the `getListingImages` endpoint for each product but that makes the page take quite a long time to load even if I only fetch 12 items at a time.

Is there any other way I can retrieve active listings with just the basic data (title, description, price, image, tags, URL) through a single endpoint with just the API key?

**Comment by Gareth064** [comment]
Hey @lubna 

When you use OAuth, people who visit your friends site will not need to sign in. Your friend will have to authorise your app to access his listings. And he will only need to do this once.

As long as the auth token is refreshed with every 90 days, the. It will never expire.

**Reply by lubna** [reply]
Thanks for the reply, that is good to know but it still a lot of extra functionality to add when both images and listings can be fetched without OAuth individually. I know the additional authentication allows you to load listings in other states so it makes sense that the extra authentication would be required to fetch draft or sold listings - however the active ones are visible both on the site (with images) and through the individual endpoints so I just thought there must be a way to get them combined without extra layers or authorisation.

**Comment by AhamSammich** [comment]
I was running into the same issues. I ended up using the findActiveListingsByShop to retrieve an array of listing_ids, then using that array with the getListings endpoint which takes the array of IDs and returns a response that can include the listing images. This way I only needed to make two requests rather than one for each product.

**Comment by mgeric** [comment]
Still, why two requests when you get back the same data, plus images the second time. Etsy should just append images to the findAllActiveListingsByShop. It's public data, not restricted by any means, only made hard to retrieve by having to do double requests.
